app/http.php

```diff
@@ -0,0 +1,366 @@
+        if (!$isProduction) {
+            $headers = \array_merge($headers, [
+                'x-open-runtimes-executor-hostname' => $executor['hostname']
+            ]);
+        }
+
+        $client->setHeaders($headers);
+        $client->setData($request->getRawPayload());
+        $client->execute($request->getURI());
+
+        $response
+            ->setStatusCode($client->getStatusCode())
+            ->setContentType(Response::CONTENT_TYPE_JSON, Response::CHARSET_UTF8)
+            ->send($client->getBody());
+    });
+
+// TODO: @Meldiron Uncomment once utopia framework supports wildcard
+/*
+App::error()
+    ->inject('utopia')
+    ->inject('error')
+    ->inject('request')
+    ->inject('response')
+    ->action(function (App $utopia, throwable $error, Request $request, Response $response) {
+        $route = $utopia->match($request);
+        logError($error, "httpError", $route);
+
+        switch ($error->getCode()) {
+            case 400: // Error allowed publicly
+            case 401: // Error allowed publicly
+            case 402: // Error allowed publicly
+            case 403: // Error allowed publicly
+            case 404: // Error allowed publicly
+            case 406: // Error allowed publicly
+            case 409: // Error allowed publicly
+            case 412: // Error allowed publicly
+            case 425: // Error allowed publicly
+            case 429: // Error allowed publicly
+            case 501: // Error allowed publicly
+            case 503: // Error allowed publicly
+                $code = $error->getCode();
+                break;
+            default:
+                $code = 500; // All other errors get the generic 500 server error status code
+        }
+
+        $output = [
+            'message' => $error->getMessage(),
+            'code' => $error->getCode(),
+            'file' => $error->getFile(),
+            'line' => $error->getLine(),
+            'trace' => $error->getTrace(),
+            'version' => App::getEnv('OPEN_RUNTIMES_PROXY_VERSION', 'UNKNOWN')
+        ];
+
+        $response
+            ->addHeader('Cache-Control', 'no-cache, no-store, must-revalidate')
+            ->addHeader('Expires', '0')
+            ->addHeader('Pragma', 'no-cache')
+            ->setStatusCode($code);
+
+        $response->json($output);
+    });
+*/
+
+/** @phpstan-ignore-next-line */
+Co\run(
+    function () use ($register) {
+        // Keep updating executors state
+        if (App::getEnv('OPEN_RUNTIMES_PROXY_HEALTHCHECK', 'enabled') === 'enabled') {
+            Timer::tick(\intval(App::getEnv('OPEN_RUNTIMES_PROXY_PING_INTERVAL', '10000')), function (int $timerId) {
+                \go(function () {
+                    fetchExecutorsState(false);
+                });
+            });
+        }
+
+        App::setMode(App::MODE_TYPE_PRODUCTION);
+
+        $server = new Server('0.0.0.0', 80, false);
```

Comment:
I couldn't find any difference in the way we init the servers here and in Appwrite. Instead of wrapping both the server and the timer in a big coroutine, we can start the server and then on server start, start the timer. See references below:

https://github.com/appwrite/appwrite/blob/master/app/http.php#L26
https://github.com/appwrite/appwrite/blob/master/app/http.php#L59-L62

Reply:
Appwrite `Server` is from `use Swoole\Http\Server;`.
Proxy `Server` is from `use Swoole\Coroutine\Http\Server;`.

Sadly there is no `->on('start'`.

<img width="494" alt="CleanShot 2022-11-02 at 08 45 05@2x" src="https://user-images.githubusercontent.com/19310830/199429457-73f8f1fa-52e0-49fa-a992-0d3c7e46178b.png">

Reply:
Also getting `Fatal error: Uncaught Swoole\Error: API must be called in the coroutine in /usr/local/app/http.php:344` if trying to start HTTP server outside of `run()`.

Reply:
I'll need to dig in and understand the differences, something feel off - not a blocker!

Reply:
Swoole official docs also recommend starting the server inside `run` in their example: https://wiki.swoole.com/#/coroutine/http_server?id=%e5%ae%8c%e6%95%b4%e7%a4%ba%e4%be%8b